backend/inspirehep/records/marshmallow/literature/bibtex.py

```diff
@@ -135,11 +135,16 @@ def get_best_publication_info(data):
             entry
             for entry in publication_info
             if entry.get("material", "publication") == "publication"
+            and entry.get("journal_title", "")
+            and entry.get("journal_volume", "")
+            and (entry.get("artid", "") or entry.get("page_start", ""))
         ]
         if not only_publications:
             return {}
 
-        return sorted(only_publications, key=len, reverse=True)[0]
+        return sorted(
+            only_publications, key=lambda item: item.get("year", float("inf"))
+        )[0]
```

Comment:
why we need different sorted?

Reply:
It should return the earliest one.

Reply:
can year be `None`? if yes I would add `[] and entry.get("year")` and we'll not need the `float(inf)`

Reply:
In that case, it throws an error; `None` type cannot be compared with `int` type. And I thought the items with a year info should come before that the ones without year info, that's why I put `float(inf)` by default.

Reply:
you can ignore the ones without `year` https://github.com/inspirehep/inspirehep/pull/1753/files#diff-33e33d63c34a6b46d8b67464e2c19087c32e16705c5589cf3aef05d824f432f0R138

Reply:
@michamos Should we ignore the publications with no year info even if there is no publication having year data? In that case, It will return `none` while there are publications having all of `journal_title`, `journal_volume`, `page_start/artid` but not `year`.